hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -807,6 +807,35 @@ def _track_subtype(self, init_values):
                 return self.env.ref("hr_timesheet_sheet.mt_timesheet_approved")
         return super()._track_subtype(init_values)
 
+    # ------------------------------------------------
+    # to prevent sending emails to the reviewer
+    # ------------------------------------------------
+
+    def _notify_compute_recipients(self, message, msg_vals):
+        """Overrides the _notify_compute_recipients method,
+
+        preventing automatic send emails to partners who have the right
+        to make a review, when the timesheet state is changes.
+        """
+        timesheet_sheet = self.browse(message.res_id)
+        recipient_data = super(Sheet, self)._notify_compute_recipients(
+            message, msg_vals
+        )
+        if timesheet_sheet.company_id.hr_timesheet_reviewer_no_email:
+            possible_reviewers = timesheet_sheet._get_possible_reviewers().mapped(
+                "partner_id"
+            )
+            sanitized_recipient_partners = []
+            for recipient in recipient_data["partners"]:
+                if (
+                    recipient["id"] not in possible_reviewers.ids
+                    or recipient["notif"] != "email"
```

Comment:
Why are you filtering out notification type here?
You should override [this](https://github.com/odoo/odoo/blob/59669e9943158e51dcbb9ae69ad758df8f7c7976/addons/mail/models/mail_thread.py#L2227) function instead in case you need to filter email notifications only

Reply:
fixed